--- a/Makefile.PL
+++ b/Makefile.PL
@@ -104,18 +104,7 @@
 
 sub ssleay {
     my $prefix = find_openssl_prefix();
-    my $exec   = find_openssl_exec($prefix);
-    unless (-x $exec) {
-        print <<EOM;
-*** Could not find OpenSSL
-    If it's already installed, please set the OPENSSL_PREFIX environment
-    variable accordingly. If it isn't installed yet, get the latest version
-    from http://www.openssl.org/.
-EOM
-        exit 0; # according http://wiki.cpantesters.org/wiki/CPANAuthorNotes this is best-practice when "missing library"
-    }
-    check_openssl_version($prefix, $exec);
-    my $opts = ssleay_get_build_opts($prefix, $exec);
+    my $opts = ssleay_get_build_opts($prefix);
     my %args = (
         CCCDLFLAGS => $opts->{cccdlflags},
         OPTIMIZE => $opts->{optimize},
@@ -133,7 +122,7 @@
 sub maybe_quote { $_[0] =~ / / ? qq{"$_[0]"} : $_[0] }
 
 sub ssleay_get_build_opts {
-    my ($prefix, $exec) = @_;
+    my ($prefix) = @_;
 
     my $opts = {
         lib_links  => [],
